### PR TITLE
Add AaveAdapter implementation and test cases

### PR DESCRIPTION
**Pull request by gvladika**
AaveAdapter implementation according to current IAdapter. 
Test cases implemented for 4 basic scenarios (invest, redeem, repay, borrow) and for both ERC-20 and native token:

  AaveAdapter
    ✓ should deposit DAI to Aave pool (37902ms)
    ✓ should withdraw DAI from Aave pool (4418ms)
    ✓ should borrow USDC from Aave pool (35513ms)
    ✓ should repay USDC to Aave pool (1548ms)
    ✓ should deposit ETH/Matic to Aave pool (26360ms)
    ✓ should withdraw ETH/Matic from Aave pool (2604ms)
    ✓ should borrow ETH/Matic from Aave pool (6103ms)
    ✓ should repay ETH/Matic to Aave pool (194ms)

TODO: test edge cases, reverts, more complex scenarios etc.

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -10,13 +12,35 @@ task("accounts", "Prints the list of accounts", async () => {
...
+  networks: {
+    hardhat: {
+      forking: {
+        url: `https://polygon-mainnet.infura.io/v3/${INFURA_API_KEY}`,
+        blockNumber: 15937268,
+      },
+    },
+  },
+  solidity: {
+    version: "0.7.0",
```
Comment (tarunbhm) [inline hardhat.config.js]: Let's use latest version of solidity - 0.8.5, Let me know if you think it will create issues. Basically 0.8.x handles math operations overflow itself. So we won't need SafeMath.sol

File: contracts/IAdapter.sol [hunk 2]
```diff
@@ -7,39 +7,40 @@ interface IAdapter {
...
-        address[] calldata tokens, 
-        uint256[] calldata amounts, 
-        address to, 
+        address tokenToBurn,
+        address[] calldata tokens,
+        uint256[] calldata amounts,
+        address to,
         uint8 ethIndex
-    ) external returns (uint256[] memory amountsTransferred);
+    ) external payable returns (uint256[] memory amountsTransferred);
```
Comment (tarunbhm) [inline contracts/IAdapter.sol]: I think this one does not need to be `payable`

File: contracts/AaveAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        return amountsTransferred;
+    }
+
+    /**
+     * @dev transfer ETH/Matic to an address, revert if it fails.
+     * @param to recipient of the transfer
+     * @param value the amount to send
+     */
+    function _safeTransferETH(address to, uint256 value) internal {
+        (bool success, ) = to.call{value: value}(new bytes(0));
```
Comment (tarunbhm) [inline contracts/AaveAdapter.sol:177]: may be better to use `address payable x = address(to); x.send(value);`. What do you think?

File: test/rebalancer-test.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    aaveProvider = await ethers.getContractAt(
+      "IProtocolDataProvider",
+      "0x7551b5D2763519d4e37e8B81929D336De671d46d",
+      testSigner
+    );
+
+    // deploy Aave adapter
+    const AaveAdapter = await ethers.getContractFactory("AaveAdapter");
+    aaveAdapter = await AaveAdapter.deploy();
+    await aaveAdapter.deployed();
```
Comment (tarunbhm) [inline test/rebalancer-test.js]: After deploying it let's create's another instance of this contract with our `IAdapter` interface artifact. This way we will be sure that the contract implements the interface as intended. If required we can make changes to function modifier in interface definition.

**Review by tarunbhm — COMMENTED** [review]
Please use `0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee` instead of zero address for ETH

File: hardhat.config.js (hunk 1 above)
Comment (gvladika) [reply hardhat.config.js]: I see you switched to 0.8.0 so I did as well.

File: test/rebalancer-test.js (hunk 4 above)
Comment (gvladika) [reply test/rebalancer-test.js]: Done

File: contracts/IAdapter.sol (hunk 2 above)
Comment (gvladika) [reply contracts/IAdapter.sol]: True, same for borrow
